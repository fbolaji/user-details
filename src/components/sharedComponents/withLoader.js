import React from "react";
import PropTypes from 'prop-types';

export const withLoader = (Component) => ({ loading, ...props }) => {
  return loading ? (
    <div data-testid="test-loader" className="loader-overlay">
      <div className="loader-circle-wrap">
        <div className="loader-circle" />
      </div>
    </div>
  ) : (<Component {...props} />)
};

  withLoader.propTypes = {
    isLoading: PropTypes.bool.isRequired,
  };

export default withLoader;
