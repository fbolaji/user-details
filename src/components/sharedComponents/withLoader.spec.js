import React from 'react';
import { render, cleanup } from '@testing-library/react';
import withLoader from './withLoader';

const MockApp = () => {
	return (
			<div>
				Component loaded
			</div>
		);
};

const MockWithHOC = withLoader(MockApp);

describe('withLoader', () => {
	afterEach(cleanup);

	it('should take component with props and if loading is true show spinnng icon while loading', () => {
		const { container } = render(<MockWithHOC loading={true} />);

		expect(container).toMatchSnapshot();
	});

	it('should take component with props and if loading is false return component', () => {
		const { container, getByText } = render(<MockWithHOC loading={false} />);

		expect(getByText(/Component loaded/i)).toBeInTheDocument();
		//expect(container.innerHTML).toMatch("Component loaded")
	});
});