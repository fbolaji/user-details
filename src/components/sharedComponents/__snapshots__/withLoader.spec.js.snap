// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoader should take component with props and if loading is true show spinnng icon while loading 1`] = `
<div>
  <div
    class="loader-overlay"
    data-testid="test-loader"
  >
    <div
      class="loader-circle-wrap"
    >
      <div
        class="loader-circle"
      />
    </div>
  </div>
</div>
`;

exports[`withLoader should take component with props and if loading is true, show spinnng icon while loading 1`] = `
<div>
  <div
    class="loader-overlay"
    data-testid="test-loader"
  >
    <div
      class="loader-circle-wrap"
    >
      <div
        class="loader-circle"
      />
    </div>
  </div>
</div>
`;

exports[`withLoader should take component with props and show spinnng while loading 1`] = `
<div>
  <div
    class="loader-overlay"
    data-testid="test-loader"
  >
    <div
      class="loader-circle-wrap"
    >
      <div
        class="loader-circle"
      />
    </div>
  </div>
</div>
`;
